server.go

```diff
@@ -234,3 +235,96 @@ func (b *backend) Move(r *http.Request, dest *internal.Href, overwrite bool) (cr
 	}
 	return created, err
 }
+
+// BackendSuppliedHomeSet represents either a CalDAV calendar-home-set or a
+// CardDAV addressbook-home-set. It should only be created via
+// `caldav.NewCalendarHomeSet()` or `carddav.NewAddressbookHomeSet()`. Only to
+// be used server-side, for listing a user's home sets as determined by the
+// (external) backend.
+type BackendSuppliedHomeSet interface {
+	GetXMLName() xml.Name
+}
+
+// UserPrincipalBackend can determine the current user's principal URL for a
+// given request context.
+type UserPrincipalBackend interface {
+	CurrentUserPrincipal(ctx context.Context) (string, error)
+}
+
+type ServeUserPrincipalOptions struct {
+	UserPrincipalPath string
+	HomeSets          []BackendSuppliedHomeSet
+}
+
+// ServeUserPrincipal replies to discovery requests. It returns false if the
+// request wasn't handled.
+func ServeUserPrincipal(w http.ResponseWriter, r *http.Request, options ServeUserPrincipalOptions) bool {
+	if r.URL.Path == "/.well-known/carddav" || r.URL.Path == "/.well-known/caldav" {
```

Comment:
Hm. I'm not a fan of hardcoding carddav and caldav here. Can we leave that to the appropriate packages?

Reply:
hey guys, i'm using this lib to serve carddav and i wonder why the restrictions on the urls !! is it defined in the rfc or what ?
Cause it's making it a bit harder to integrate this to an already developped application.
Thanks both of you.

Reply:
Please open an issue with more details. I don't really understand what you're talking about here.

Reply:
okey i'll delete this comment and open an issue